src/spikeinterface/core/baserecording.py

```diff
@@ -242,34 +242,34 @@ def get_total_memory_size(self) -> int:
 
     def get_traces(
         self,
-        segment_index: Union[int, None] = None,
-        start_frame: Union[int, None] = None,
-        end_frame: Union[int, None] = None,
-        channel_ids: Union[Iterable, None] = None,
-        order: Union[str, None] = None,
-        return_scaled=False,
-        cast_unsigned=False,
+        segment_index: int | None = None,
+        start_frame: int | None = None,
+        end_frame: int | None = None,
+        channel_ids: list | np.array | None = None,
+        order: "C" | "F" | None = None,
+        return_scaled: bool = False,
+        cast_unsigned: bool = False,
     ):
         """Returns traces from recording.
 
         Parameters
         ----------
-        segment_index : Union[int, None], default: None
+        segment_index : int | None, default: None
             The segment index to get traces from. If recording is multi-segment, it is required, default: None
-        start_frame : Union[int, None], default: None
+        start_frame : int | None, default: None
             The start frame. If None, 0 is used, default: None
-        end_frame : Union[int, None], default: None
+        end_frame : int | None, default: None
             The end frame. If None, the number of samples in the segment is used, default: None
-        channel_ids : Union[Iterable, None], default: None
+        channel_ids : list | np.array | None, default: None
             The channel ids. If None, all channels are used, default: None
-        order : Union[str, None], default: None
-            The order of the traces ("C" | "F"). If None, traces are returned as they are, default: None
-        return_scaled : bool, default: None
+        order : "C" | "F" | None, default: None
```

Comment:
This should be a literal

Reply:
It can be a Literal, but as of 3.10 it doesn't have to be a Literal. We can list them as strings, but type checkers will complain. Again I think it looks cleaner without, but if we want to always use Literal we can. As a user I think it is cleaner to see without the surrounding `Literal[a | b]`, but some of these are more personal aesthetics.

Reply:
Interesting. Can you point the documentation of this to me? The change in 3.10 I mean.

I honestly don't care that much about it, if you like it that was it is fine for me. As you mentioned above, our use of typing is an aid to documentation.

Reply:
Sorry I meant use of the `|` was 3.10. Typing always accepted strings due to how they are parsed. So it used to be that you could do typing with:

```python
def func(a: "int | float " = 5)
```
So the use of strings isn't new to 3.10, but the fact we can use the pipe without needing to in case of strings. So if we want to eventually go to type checking then I think we would need to use `Literal` for the type checker to properly parse. If we are just using this as docs then it doesn't matter. I'll do either. If there is any chance we would consider type checking I would say let's add in the Literal now so it's less burden later, but if we are never going to type check then I like without Literal.

Reply:
I don't undesrstand. Literal means that there is a fixed set of options from which you want could choose: ["C", "F", None] are the set of possible values.

Does "C" | "F" | None now has the same semantics as Literal["C", "D", None]? 
That was the documentation I was looking for. Out of curiosity as I mentioned, if you like it that way, it is fine for me.

Reply:
Ah, reading the relevant PEPs does not spell it out explictly it seems but I think what you mention is correct.